src/main/java/com/hcl/appscan/jenkins/plugin/scanners/Scanner.java

```diff
@@ -45,18 +43,10 @@ public boolean getHasOptions() {
 	public String getTarget() {
 		return m_target;
 	}
-        
-	public boolean isRescan() {
-		return m_rescan;
-	}
-        
-	public String getScanId() {
-		return m_scanId;
-	}
 	
 	public abstract Map<String, String> getProperties(VariableResolver<String> resolver) throws AbortException;
 
-	public abstract void validateSettings(JenkinsAuthenticationProvider authProvider, Map<String, String> properties, IProgress progress) throws AbortException;
+	public abstract void validateSettings(JenkinsAuthenticationProvider authProvider, Map<String, String> properties, IProgress progress, boolean isAppScan360) throws IOException;
```

Comment:
Rather than having a validateSettings() that each scanner needs to implement as well as a validateGeneralSettings() method that each scanner needs to call, I suggest making 1 public method called validateSettings() that takes care of verifying all of the settings that apply to every scanner. There can also be an abstract method called something like validateScannerSettings() that is called at the end of validateSettings() in the Scanner class. That way, each subclass simply implements the abstract validateScannerSettings() method and there's no need to worry about base classes calling the method from the super class, etc.

Reply:
Made the changes as per suggestion.